drivers/gpio/gpio_nrfx.c

```diff
@@ -0,0 +1,467 @@
+						int access_op,
+						u32_t pin)
+{
+	return gpio_nrfx_pin_manage_callback(port, access_op, pin, true);
+}
+
+static inline int gpio_nrfx_pin_disable_callback(struct device *port,
+						 int access_op,
+						 u32_t pin)
+{
+	return gpio_nrfx_pin_manage_callback(port, access_op, pin, false);
+}
+
+static const struct gpio_driver_api gpio_nrfx_drv_api_funcs = {
+	.config = gpio_nrfx_config,
+	.write = gpio_nrfx_write,
+	.read = gpio_nrfx_read,
+	.manage_callback = gpio_nrfx_manage_callback,
+	.enable_callback = gpio_nrfx_pin_enable_callback,
+	.disable_callback = gpio_nrfx_pin_disable_callback
+};
+
+/**
+ * @brief Function for getting pins that triggered level interrupt.
+ *
+ * @param port Pointer to GPIO port device.
+ *
+ * @return Bitmask where 1 marks pin as trigger source.
+ */
+static u32_t check_level_trigger_pins(struct device *port)
+{
+	if (port == NULL) {
+		return 0;
+	}
+	struct gpio_nrfx_data *data = get_port_data(port);
+
+	u32_t port_in = nrf_gpio_port_in_read(get_port_cfg(port)->port);
+
+	/* Extract which pins after inversion, have logic level same as
+	 * interrupt trigger level.
+	 */
+	u32_t pin_states = ~(port_in ^ data->inverted ^ data->active_level);
+
+	/* Take into consideration only pins that were configured to
+	 * trigger interrupts and have callback enabled.
+	 */
+	u32_t out = pin_states & data->int_en & data->pin_int_en;
+
+	/* Exclude pins that trigger interrupts by edge. */
+	out &= ~data->trig_edge & ~data->double_edge;
+
+	return out;
+}
+
+static inline void fire_callbacks(struct device *port, u32_t pins)
+{
+	_gpio_fire_callbacks(&get_port_data(port)->callbacks, port, pins);
+}
+
+#ifdef CONFIG_GPIO_NRFX_P0
+DEVICE_DECLARE(gpio_nrfx_p0);
+#endif
+#ifdef CONFIG_GPIO_NRFX_P1
+DEVICE_DECLARE(gpio_nrfx_p1);
+#endif
+
+static void gpiote_event_handler(void)
+{
+	u32_t int_trigger_pins[GPIO_COUNT];
+	u32_t active_level_triggers;
+
+#ifdef CONFIG_GPIO_NRFX_P0
+	int_trigger_pins[0] =
+		check_level_trigger_pins(DEVICE_GET(gpio_nrfx_p0));
+#endif
+#ifdef CONFIG_GPIO_NRFX_P1
+	int_trigger_pins[1] =
+		check_level_trigger_pins(DEVICE_GET(gpio_nrfx_p1));
+#endif
+
+
+	/* Handle interrupt from GPIOTE channels. */
+	for (size_t i = 0; i < GPIOTE_CH_NUM; i++) {
+		nrf_gpiote_events_t evt =
+			offsetof(NRF_GPIOTE_Type, EVENTS_IN[i]);
+
+		if (nrf_gpiote_event_is_set(evt)) {
+			u32_t abs_pin = nrf_gpiote_event_pin_get(i);
+			/* Divide absolute pin number to port and pin parts. */
+			int_trigger_pins[abs_pin / 32] |= BIT(abs_pin % 32);
+			nrf_gpiote_event_clear(evt);
+		}
+	}
+
+#ifdef CONFIG_GPIO_NRFX_P0
+	if (int_trigger_pins[0]) {
+		fire_callbacks(DEVICE_GET(gpio_nrfx_p0), int_trigger_pins[0]);
+	}
+#endif
+#ifdef CONFIG_GPIO_NRFX_P1
+	if (int_trigger_pins[1]) {
+		fire_callbacks(DEVICE_GET(gpio_nrfx_p1), int_trigger_pins[1]);
+	}
+#endif
+
+#ifdef CONFIG_GPIO_NRFX_P0
+	int_trigger_pins[0] =
+		check_level_trigger_pins(DEVICE_GET(gpio_nrfx_p0));
```

Comment:
There is no need to update `int_trigger_pins` here. Just do:
```
	active_level_triggers =
#ifdef CONFIG_GPIO_NRFX_P0
		check_level_trigger_pins(DEVICE_GET(gpio_nrfx_p0)) |
#endif
#ifdef CONFIG_GPIO_NRFX_P1
		check_level_trigger_pins(DEVICE_GET(gpio_nrfx_p1)) |
#endif
		0;
```
And this should be done only when the PORT event is set. Then also the `active_level_triggers` can be local to the related `if` block.